frontend/app/src/connection/DefaultStreamlitEndpoints.ts

```diff
@@ -142,9 +167,21 @@ export class DefaultStreamlitEndpoints implements StreamlitEndpoints {
     fileUrl: string,
     sessionId: string
   ): Promise<void> {
+    let headers: Record<string, string> = {}
+    if (this.jwtHeader !== undefined) {
+      headers[this.jwtHeader.jwtHeaderName] = this.jwtHeader.jwtHeaderValue
+    }
+
+    if (this.fileUploadClientConfig) {
+      headers = {
+        ...headers,
+        ...this.fileUploadClientConfig.headers,
+      }
+    }
```

Comment:
These lines seem to be duplicate with the ones in `buildFileUploadUrl` [here](https://github.com/streamlit/streamlit/pull/9727/files#diff-87a5e69b1165fc1aab4cf6c886c04857851089df4f169a5c9927528a86b89c57R97). It might make sense to move this into a shared function, e.g. `setHeaders` or similar

Reply:
done, moved it to separate function, it will be even more simple when we remove `jwtHeader` in near future after this changes